coremltools/converters/mil/frontend/torch/ops.py

```diff
@@ -781,11 +779,21 @@ def flatten(context, node):
         raise ValueError(
             "Start must be before end. (start, end) == ({}, {})".format(start, end_val)
         )
-    for dim in dims[start : end + 1]:
-        total *= dim
-    dims = dims[:start] + [total] + dims[end + 1 :]
+    if start == 0 and end == len(dims) - 1:
```

Comment:
I'm not really following the logic here. Could you add some comments. More descriptive variables names would help too.

Reply:
This logic is for generate simple structure when start == 0 and end == -1.
But in many cases, start is not zero since first dimension is batch.
So I remove this logic.